src/main/java/uk/gov/justice/digital/delius/service/ReferralService.java

```diff
@@ -115,6 +115,7 @@ public ReferralEndResponse endNsiReferral(final String crn, final String context
         var existingNsis = nsiService.getNsiByCodes(offenderId, sentenceId, Collections.singletonList(getNsiType(nsiMapping, contractType)))
             .map(wrapper -> wrapper.getNsis().stream()
                 // eventID, offenderID, nsi type are handled in the NSI service
+                .filter(nsi -> !nsi.getSoftDeleted())
```

Comment:
I like to suggest a breaking change; move the filter into the service call. Check with @chris-faulkner-mojdt that the endpoint added last year doesn't need deleted NSI. I would be 99% certain it does not and is an existing bug in that service.

Reply:
Will do

Reply:
Decided to proceed and will do this as a secondary change on Chris' approval